dev-libs/botan/botan-2.6.0.recipe

```diff
@@ -0,0 +1,93 @@
+SUMMARY="A cryptography library written in C++"
+DESCRIPTION="This is the best option for crypthography in C++"
+HOMEPAGE="https://botan.randombit.net/"
+COPYRIGHT="1999-2018 The Botan Authors"
+LICENSE="BSD (2-clause)"
+REVISION="1"
+SOURCE_URI="https://github.com/randombit/botan/archive/$portVersion.tar.gz"
+CHECKSUM_SHA256="819e5f697a986898ad090b89483f362c7b768db81bc313070e45e40a9cbdb52b"
+
+ARCHITECTURES="!x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="!x86_gcc2 x86"
+
+if [ "$targetArchitecture" != x86_gcc2 ]; then
+	commandSuffix=$secondaryArchSuffix
+	commandBinDir=$binDir
+else
+	commandSuffix=
+	commandBinDir=$prefix/bin
+fi
+
+PROVIDES="
+	botan$secondaryArchSuffix = $portVersion
+	lib:libbotan_2$secondaryArchSuffix = 5.6.0 compat >= 5
+	cmd:botan$commandSuffix = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libbz2$secondaryArchSuffix
+	lib:libcrypto$secondaryArchSuffix
+	lib:liblzma$secondaryArchSuffix
+	lib:libsqlite3$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	botan${secondaryArchSuffix}_devel = $portVersion
+	devel:libbotan_2$secondaryArchSuffix = 5.6.0 compat >= 5
+	"
+REQUIRES_devel="
+	haiku${secondaryArchSuffix}_devel
+	botan$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libbz2$secondaryArchSuffix
+	devel:libcrypto$secondaryArchSuffix
+	devel:liblzma$secondaryArchSuffix
+	devel:libsqlite3$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:awk
+	cmd:gcc$secondaryArchSuffix
+	cmd:g++$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:libtoolize$secondaryArchSuffix
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	cmd:python
+	"
+
+PATCH()
+{
+	sed -i -e "/^all\>/ s/-pthread -fstack-protector/-fno-stack-protector/" src/build-data/cc/gcc.txt
```

Comment:
I think we only need to remove `-pthread` now, as the stack protector is fixed.

Reply:
Need to check this, not a big sed guru ;)

Reply:
It builds fine with
```
	sed -i -e "/^all\>/ s/-pthread//" src/build-data/cc/gcc.txt
```
but maybe we can even drop the `PATCH()` function completely.

Reply:
I think they already included it in the source "all!haiku -> "-pthread"", could be this patch isn't needed anymore ...

Reply:
Should this still be needed to use sed @fbrosson ?

Reply:
(I've commented out the `PATCH()` function to make sure we can drop the `sed` command (and the whole `PATCH()` function.)

Reply:
Same here, build is still running though ...

Reply:
Looks like the PATCH() function isn't needed anymore, untill someone steps up I've removed it from the recipe, builds ok for secondaryarch (didn't check x86_64 yet so best someone would check this before merging) :)

Reply:
I kicked 3 parallel builds: botan_x86 on x86_gcc2, and botan on x86 and x86_64.
When I try to test any of these builds I get:
```
Testing ...
runtime_loader: Cannot open file libbotan-2.so.5: No such file or directory
Warning: Command '['bash', '-c', '. /wrapper-script']' returned non-zero exit status 3
Error: Test has failed - stopping.
```
We can avoid this error with this change:
```diff
 TEST()
 {
+	LIBRARY_PATH="$sourceDir${LIBRARY_PATH:+:$LIBRARY_PATH}" \
 	./botan-test
 }
```
Unfortunately, we now get a crash with it:
```
Testing ...
Testing Botan 2.6.0 (release, dated 20180410, revision git:d7d080992372cf4fbd569cce1d8cd6aa7599fa0d, distribution unspecified)
Starting tests drbg_seed:1534AB7A5ABD1248
block:
/port.recipe: line 101: 28865 Kill Thread
```
Well, anyway, the suggested change in `TEST` (i.e. inserting the line which changes `LIBRARY_PATH`) is a good thing to do.

Reply:
With the suggestion you made it seems like the tests are running ok here ... 
'''
Testing Botan 2.6.0 (release, dated 20180410, revision git:d7d080992372cf4fbd569cce1d8cd6aa7599fa0d, distribution unspecified)
  Starting tests drbg_seed:1534AC7A982411A8
  block:
  AES-128 ran 11580 tests in 157.09 msec all ok
  AES-192 ran 13500 tests in 140.75 msec all ok
  AES-256 ran 15420 tests in 172.17 msec all ok
'''

Reply:
Ending in ... 
```
Entropy source system_rng ran 9 tests in 370.67 msec 1 FAILED
  Failure 1: Entropy source system_rng lzma exception while compressing with error lzma memory allocation failed ...

 Tests complete ran 2621414 tests in 511.63 sec 1 tests failed
  Warning: Command '['bash', '-c', '. /wrapper-script']' returned non-zero exit status 1
  Error: Test has failed - stopping
```
Maybe libzma is missing for TEST?

Reply:
Good idea. We might solve this with:
```diff
 	cmd:python
 	"
 
+TEST_REQUIRES="
+	devel:liblzma$secondaryArchSuffix
+	"
+
 BUILD()
 {
```

Reply:
No luck there, still the same ...

Reply:
Well, you are the developer ;)